packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/index/ProjectTemplateIndexer.java

```diff
@@ -0,0 +1,53 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.web.application.index;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import org.eclipse.sirius.components.core.api.IEditingContext;
+import org.eclipse.sirius.components.core.api.IEditingContextSearchService;
+import org.eclipse.sirius.web.application.index.api.IEditingContextIndexer;
+import org.eclipse.sirius.web.application.project.dto.CreateProjectFromTemplateInput;
+import org.eclipse.sirius.web.domain.boundedcontexts.semanticdata.events.SemanticDataUpdatedEvent;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.event.TransactionalEventListener;
+
+/**
+ * Indexes the initial content of a project created from a template.
+ *
+ * @author gdaniel
+ */
+@Service
+public class ProjectTemplateIndexer {
+
+    private final IEditingContextSearchService editingContextSearchService;
+
+    private final IEditingContextIndexer editingContextIndexer;
+
+    public ProjectTemplateIndexer(IEditingContextSearchService editingContextSearchService, IEditingContextIndexer editingContextIndexer) {
+        this.editingContextSearchService = Objects.requireNonNull(editingContextSearchService);
+        this.editingContextIndexer = Objects.requireNonNull(editingContextIndexer);
+    }
+
+    @TransactionalEventListener
+    public void onSemanticDataUpdatedEvent(SemanticDataUpdatedEvent semanticDataUpdatedEvent) {
+        if (semanticDataUpdatedEvent.causedBy() instanceof CreateProjectFromTemplateInput) {
```

Comment:
Since initializing a project from a template doesn't go through the event processors I can't react to it the same way I react to other input.
The solution I found is to index the entire project when this event is received, but I am not sure it is the best way to achieve it.